### DISCUSSION TITLE
Some modifications about the RDK plugin to make it in the center of the window even in fullscreen mode

### DISCUSSION THREAD

**Discussion by zx23237** [jspsych/jsPsych#1333] [Ideas]
I used RDK plugins these days, and I found that the fixation is not in the center of the window.

I found the reason and add some code when using the plugin.

1. Due to jsPsych CSS set the visible width to 95% of window‘s width, the **default centerX of aperture** used in the original RDK plugin, which is window.innerwidth/2, should be **window.innerwidth*0.95/2** to locate both the aperture and fixation in the center of the window.

2. In fullscreen mode, as the centerY of aperture (window.innerheigh/2) was get as the experiment window opened, instead of the window of fullscreen, there might be slight deviation of centerY get from window.innerheigh/2. My solution is to refresh the width and height of the window at the beginning of each trial, by **using on_start(trial) function** .

The code I added for the trial of presenting RDK is as follow:
```
on_start(trial) {
        if (window.outerHeight == screen.height) {
            if (window.outerHeight == window.innerHeight) {
                var center_y = window.outerHeight*0.5;
            }else{
                var center_y = window.innerHeight*0.5;
            }
        }else{
            var center_y = window.innerHeight*0.5;
        }
        var center_x = window.innerWidth*0.475;
        trial.aperture_center_x = center_x;
        trial.aperture_center_y = center_y
    }
```

**Comment by becky-gilbert** [comment]
Hi @zx23237, thanks very much for sharing these improvements! I've opened an issue so that we can track this problem and hopefully fix it soon: #1355. It'd be great if you wanted to submit a pull request with your suggested changes to the RDK plugin (but no problem if not).

**Reply by zx23237** [reply]
Thanks @becky-gilbert! 

My experiment trial is to first show a fixation cross at the center and then show one aperture of RDK at the center of screen. 

I used text "+" to make the fixation with html-keyboard-response plugin, which should be at the center of the screen in default. But the center of a single RDK in circle shape did not appear at the same position as the fixation cross. 

Furthermore, when I tried to use the RDK plugin in Chrome or Edge for Window 10, the scrollbar would show, which indicates that the canvas height and width of RDK plugin might not be suit to the visible area of the browser...

**Reply by becky-gilbert** [reply]
Ok, thanks @zx23237. Could you let me know which version of jsPsych you're using for this experiment? I ask because in the last release, [version 6.2](https://github.com/jspsych/jsPsych/releases/tag/v6.2.0), we thought we'd fixed an issue related to the scroll bars in the RDK plugin (tracked in issue #881).

**Reply by zx23237** [reply]
I used the version 6.1.0 RDK plugin for my experiment. Thanks @becky-gilbert !

**Reply by becky-gilbert** [reply]
Aha, ok great, thanks for letting me know! In that case I've closed the issue that I opened because I think we resolved it in the 6.2 release. If you get a chance to try out the RDK plugin in v6.2.0, I'd be interested to know whether you're happy with the fixation centering.  

Thanks again for reporting this problem with the 6.1.0 version of the RDK plugin, and for sharing your solution! 😃
